### PR TITLE
Rename task now deletes processed documents.

### PR DESCRIPTION
**Pull request by fabianvf**
Clean it up a little too. @chrisseto @erinspace

### CODE REVIEW

File: scripts/rename.py [hunk 1]
```diff
@@ -1,49 +1,47 @@
 import logging
 
-from cqlengine import Token
+
```
Comment (chrisseto) [inline scripts/rename.py]: Extra newline

File: scripts/rename.py [hunk 2]
```diff
@@ -1,49 +1,47 @@
...
-                    'timestamps': doc.timestamps
-                })
-            except Exception as e:
-                logger.exception(e)
-        page = list(query.filter(pk__token__gt=Token(page[-1].pk)))
-
-
-def rename(source, target):
+def rename(source, target, dry=True):
+    assert source != target
```
Comment (chrisseto) [inline scripts/rename.py]: Maybe add a useful error message here?
```python
assert condition, 'some error'
```

File: scripts/rename.py [hunk 3]
```diff
@@ -1,49 +1,47 @@
...
+                'docID': doc.docID,
+                'source': target,
+                'filetype': doc.filetype,
+                'timestamps': doc.timestamps,
+                'versions': doc.versions
+            })
+            if not dry:
+                process_raw(raw)
+                process_normalized(normalize(raw, raw['source']), raw)
+            else:
```
Comment (chrisseto) [inline scripts/rename.py]: Make sure to log even when not in a dry run

File: scripts/rename.py [hunk 4]
```diff
@@ -1,49 +1,47 @@
...
+                logger.info('Processing document from {} with id {}'.format(source, raw['docID']))
         except Exception as e:
             logger.exception(e)
             exceptions.append(e)
+        else:
+            if not dry:
+                doc.delete()
+                es.delete(index=settings.ELASTIC_INDEX, doc_type=source, id=raw['docID'])
+            else:
+                logger.info('Deleting document from {} with id {}'.format(source, raw['docID']))
```
Comment (chrisseto) [inline scripts/rename.py]: See above

File: scripts/renormalize.py [hunk 5]
```diff
@@ -1,43 +1,32 @@
...
-
-def main():
-    for raw in document_generator():
+def renormalize(sources=None):
+    count = 0
+    exceptions = []
+    for doc in documents(sources):
+        count += 1
         try:
+            raw = RawDocument({
```
Comment (chrisseto) [inline scripts/renormalize.py]: Maybe we should make this into a classmethod?
```python
RawDocument.from_model(...)
# or
RawDocumentModel.to_raw_doc(...)
```

File: scripts/util.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+from scrapi.database import _manager
+from scrapi.processing.cassandra import DocumentModel
+
+_manager.setup()
+
+
+def documents(sources=None):
+    q = DocumentModel.objects.all().limit(1000)
+    querysets = (q.filter(source=source) for source in sources) if sources else [q]
```
Comment (chrisseto) [inline scripts/util.py]:
```python
querysets = (q.filter(source=source) for source in sources) or (q, )
```

File: scripts/util.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+from cqlengine import Token
+
+from scrapi.database import _manager
+from scrapi.processing.cassandra import DocumentModel
+
+_manager.setup()
+
+
+def documents(sources=None):
```
Comment (chrisseto) [inline scripts/util.py]:
```python
def documents(*sources):  # ?
   ...
```

File: scripts/renormalize.py (hunk 5 above)
Comment (fabianvf) [inline scripts/renormalize.py]: yeah, not a bad idea. Probably should be able to serialize the model to raw/normalized document, and the documents to json...

**Comment by fabianvf** [comment]
@chrisseto ready for merge?

File: scripts/renormalize.py [hunk 8]
```diff
@@ -1,43 +1,32 @@
...
-                    logger.exception(e)
-        page = list(query.filter(pk__token__gt=Token(page[-1].pk)))
-
-
-def main():
-    for raw in document_generator():
+def renormalize(sources=None):
+    count = 0
+    exceptions = []
+    for doc in documents(*sources):
```
Comment (chrisseto) [inline scripts/renormalize.py]: `*None` will error

**Pull request merged**